src/js/components/Home/Friends.jsx

```diff
@@ -1,24 +1,37 @@
-import Image from '../../../images/mountain.jpg';
-
 import styles from './Friends.module.css';
 import FriendInfo from './FriendInfo';
+import { useEffect, useState } from 'react';
+import { getFriends } from '../../API/friends/methods';
+
+const Friends = ({ searchTerm }) => {
+  const [searchResult, setSearchResult] = useState([]);
+
+  useEffect(() => {
+    getFriends().then((friends) => {
```

Comment:
Nie powinniśmy wysyłać zapytania o przyjaciół po każdej zmianie tesktu w searchabarze.
W obecnym momencie po wprowadzeniu każdej nowej litery w textfield poleci request o przyjaciół pomimo, że tak naprawdę interesuje nas tylko ostatni z wyników. 
Mamy 2 rozwiązania:
- pobieramy listę znajomych jednokrotnie i zapisujemy gdzieś w stanie, a następnie jedynie filtrujemy tych znajomych (w tym przypadku będziemy musieli w jakiś sposób odświeżać dane o użytkownikach co określony czas - problem może rozwiązać biblioteka do cachowania SWR)
- używamy debounce, żeby pobrać z backendu listę znajomych przefiltrowaną  po naszym searchphrase, ale ograniczamy ilość możliwych wywołań tej funkcji w krótkim okresie czasu - debounce (debounce z Lodasha albo równoważny)
`
	const debouncedSave = useCallback(
		debounce(nextValue => saveToDb(nextValue), 1000),
		[], // will be created only once initially
	);
`
	`

Reply:
Skorzystałam z tego pierwszego rozwiązania. Na razie nie użyłam tej biblioteki SWR, bo nie mam jak pobrać danych z backendu. Myślę, że można to dodać jak już będzie odpowiedni endpoint.